tests/end_to_end_tests/managed_packages_tests/test_packages.py

```diff
@@ -237,18 +239,15 @@ def test_packages(
     else:
         raise Exception("Starting agent message is not found.")
 
-    # On Ubuntu 22.04 agent can use its OpenSSL 3 version.
-    if distro_name == "ubuntu2204":
-        assert "OpenSSL version=OpenSSL 3.0.2" in starting_agent_message
+    m = re.search(r"OpenSSL version_number=(\d+)", starting_agent_message)
+    openssl_version = int(m.group(1))
 
-    # On Ubuntu 20.04 it can be OpenSSL 1.1.1
-    elif distro_name == "ubuntu2004":
-        assert "OpenSSL version=OpenSSL 1.1.1" in starting_agent_message
+    if isinstance(target_distro.expected_openssl, list):
+        expected_openssl_versions = target_distro.expected_openssl
+    else:
+        expected_openssl_versions = [target_distro.expected_openssl]
 
-    # On Ubuntu 14.04 there's no "modern" version OpenSSl, but it still has to use 1.1.1 that is shipped with
-    # the package.
-    elif distro_name == "ubuntu1404":
-        assert "OpenSSL version=OpenSSL 1.1.1" in starting_agent_message
+    assert openssl_version in expected_openssl_versions
```

Comment:
So we do a partial "contains" check here.

I assume this should be sufficient as long as the input string is not some loose partial sub-string which could still result in this assertion passing, but  possibly not doing what we want (e.g. if someone accidentally used just ``OpenSSL version`` or similar for the ``expected_openssl`` parameter).

Reply:
This has to use integer version value from [ssl module ](https://docs.python.org/3/library/ssl.html#ssl.OPENSSL_VERSION_NUMBER).  It have to get expected values from the agent starting message from the `agent.log` file. This message now has new field [version_number](https://github.com/scalyr/scalyr-agent-2/blob/3daaede149eac0ec0171e68d96d0c3980ef66dbb/scalyr_agent/util.py#L2171) which also shows numeric representation of the OpenSSL. For more info search for `OPENSSL_VERSION_NUMBER` in https://www.openssl.org/docs/man3.0/man3/OPENSSL_VERSION_NUMBER.html

Reply:
PS: This "contains" check does not a partial comparison. It does 1 to 1 match of integer versions of OpenSSL.